peekingduck/pipeline/nodes/dabble/statistics.py

```diff
@@ -0,0 +1,221 @@
+        identity (:obj:`str`): **default=null** |br|
+            Accepts ``<target_attr>`` of types :obj:`int` or :obj:`float`, and returns the
+            same value.
+        length (:obj:`str`): **default=null** |br|
+            Accepts ``<target_attr>`` of types :obj:`List[Any]` or :obj:`Dict[str, Any]`,
+            and returns its length.
+        minimum (:obj:`str`): **default=null** |br|
+            Accepts ``<target_attr>`` of types :obj:`List[float | int]` or
+            :obj:`Dict[str, float | int]`, and returns the minimum element within for the current
+            frame. Not to be confused with the ``cum_min`` output data type, which represents the
+            cumulative minimum over time.
+        maximum (:obj:`str`): **default=null** |br|
+            Accepts ``<target_attr>`` of types :obj:`List[float | int]` or
+            :obj:`Dict[str, float | int]`, and returns the maximum element within for the current
+            frame. Not to be confused with the ``cum_max`` output data type, which represents the
+            cumulative maximum over time.
+        cond_count (:obj:`str`): **default=null** |br|
+            Accepts ``<target_attr>`` of types :obj:`List[float | int | str]`, and checks if each
+            element in the list fulfils the condition described by ``<num_comparison>`` or
+            ``<str_comparison>``. The number of elements that fulfil the condition are counted
+            towards ``<current result>``.
+    """
+
+    def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
+        super().__init__(config, node_path=__name__, **kwargs)
+        self.cum_avg, self.cum_min, self.cum_max = 0.0, float("inf"), float("-inf")
+        self.num_iter = 0
+        all_funcs = {
+            "cond_count": self.cond_count,
+            "identity": self.identity,
+            "length": self.length,
+            "minimum": self.minimum,
+            "maximum": self.maximum,
+        }
+        self.stats = utils.Stats(OPS)
+        self.data_type, self.keys = self.stats.prepare_data(all_funcs)
+
+    def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+        """Calculates the average, minimum and maximum of a single variable of interest over time.
+
+        Args:
+            inputs (dict): Dictionary with all available keys.
+
+        Returns:
+            outputs (dict): Dictionary with keys "cum_avg", "cum_min" and "cum_max".
+        """
+
+        self.curr = self.stats.get_curr_result(inputs[self.data_type], self.keys.copy())
+
+        # if no detections in this frame, return stats from previous detections
+        if not self.curr:
+            return {
+                "cum_avg": self.cum_avg,
+                "cum_min": self.cum_min,
+                "cum_max": self.cum_max,
+            }
+
+        self._update_stats(self.curr)
+
+        return {
+            "cum_avg": self.cum_avg,
+            "cum_min": self.cum_min,
+            "cum_max": self.cum_max,
+        }
```

Comment:
Realised lines 190-202 can be shortened to:
```
if self.curr:
    self._update_stats(self.curr)

return {
    "cum_avg": self.cum_avg,
    "cum_min": self.cum_min,
    "cum_max": self.cum_max,
}
```

Reply:
Good catch, fixed.